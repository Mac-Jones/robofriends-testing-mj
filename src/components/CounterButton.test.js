import React from 'react';
import { shallow } from 'enzyme';
import CounterButton from './CounterButton';

it('Expect to render CounterButton', () => {
	expect(shallow(<CounterButton />)).toMatchSnapshot();
});

// #counter or [id='counter']
it('correnctly increments the counter', () => {
	const mockColor = 'red';
	const wrapper = shallow(<CounterButton color={mockColor} />);
	expect(wrapper).toMatchSnapshot();
	wrapper.find('#counter').simulate('click');
	wrapper.find('#counter').simulate('click');
	expect(wrapper.state()).toEqual({ count: 2 });
	wrapper.find('#counter').simulate('click');
	expect(wrapper.state()).toEqual({ count: 3 });
	wrapper.find('#counter').simulate('keypress');
	expect(wrapper.state()).toEqual({ count: 3 });
	expect(wrapper.props().color).toEqual('red');
});
