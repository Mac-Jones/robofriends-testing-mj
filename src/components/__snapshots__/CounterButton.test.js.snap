// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Expect to render CounterButton 1`] = `
<button
  id="counter"
  onClick={[Function]}
>
  Count: 
  0
</button>
`;

exports[`correnctly increments the counter 1`] = `
<button
  color="red"
  id="counter"
  onClick={[Function]}
>
  Count: 
  0
</button>
`;
